docs/src/5-prestashop-billing/2-concepts/README.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Concepts
+---
+
+[[toc]]
+
+# Concepts
+
+## Context
+
+The context is a central piece of data in the Prestashop Billing integration. We mostly use it to gather several key informations about your product, as well as the shop that uses it. Most of it comes from the `module-lib-billing` php package, and you can even see more informations about how it's constructed by browsing its [source code](https://github.com/PrestaShopCorp/module-lib-billing/blob/main/src/Presenter/BillingPresenter.php). Make sure you understand well its implementation [here](../3-tutorial/README.md#inject-the-prestashop-billing-context) and [here](../3-tutorial/README.md#edit-the-template-file) and how they interact.
+
+:::tip
+If you're not sure about some piece of code, do not hesitate to contact our team directly, we're here to help.
+:::
+
+## Events
+
+### Webhook
+
+When a change happens on the PrestaShop Billing API, you will be notified via a [webhook system](https://en.wikipedia.org/wiki/Webhook).
+These events follow the subscription lifecycle below, and are detailed in our [reference](../5-references/1-webhook/README.md#).
+
+### Subscription lifecycle
+
+You can see here the lifecycle of a subscription for different use cases.
+
+Subscription with trial
+![lifecycle subscription with trial](/assets/images/3-webhook-events/lifecycle_subscription_with_trial.jpg)
+Subscription without trial
+![lifecycle subscription without trial](/assets/images/3-webhook-events/lifecycle_subscription_without_trial.jpg)
+Subscription during trial
+![lifecycle subscription during trial](/assets/images/3-webhook-events/lifecycle_cancellation_during_trial.jpg)
+Plan change
+![lifecycle plan change](/assets/images/3-webhook-events/lifecycle_plan_change.jpg)
+
+## Subscription
+
+A subscription allows Prestashop Billing to collect periodically the payment of your customer. At each subscription renewal, an invoice is generated. A subscription is linked to your module, a shop and a customer. A shop can only have one subscription to your module.
+
+## Subscription Item
+
+The subscription items are the pricing applied to the subscription with sometimes another information about the usage or the quantity which allow us to calculate the amount of this subscription for a billing period. The subscription item id matches the pricing id.
+
+## Billing Period
+
+The interval between 2 subscription renewals. It defines when an invoice will be generated. In our case it’s a month or a year.
+
+## Shop
+
+A shop is a PrestaShop store.
+
+## Organization
+
+An organization is the person or the legal entity which owns the shop.
+
+## Payment Method
+
+A payment method designates the way a payment can be collected. Currently, Prestashop Billing offers payment via Credit Card or Paypal.
+
+## Pricing models
+
+Prestashop Billing currently supports two types of pricing models:
+
+- Flat-Fee
+- Stairstep
```

Comment:
Since there is no single way of writing stairstep and since Chargebee uses "stairstep", we decided that it should be the proper way to write it (https://www.chargebee.com/docs/2.0/plans.html#pricing-models)

There is another one below, as well as some Flat-Fee that should be flat fee that I'm fixing right now